### DISCUSSION TITLE
feat: Add OpenSearch as a vector store to open-webui

### DISCUSSION THREAD

**Discussion by dtaivpp** [open-webui/open-webui#5863] [General]
**Is your feature request related to a problem? Please describe.**
open-webui should support other vector stores such as OpenSearch for RAG. I am really putting this here to see if there is any objections to adding another vector store. I am happy to contribute this so long as it aligns with the project. 

**Describe the solution you'd like**
To start with, I think a straightforward adaptor should be implemented that allows users to use OpenSearch as a RAG backend with open-webui. 

**Describe alternatives you've considered**
Not supporting would be easier but it's always nice to have alternatives. 

**Additional context**

**Comment by tjbck** [comment]
Feel free to make a PR! https://github.com/open-webui/open-webui/tree/dev/backend/open_webui/apps/retrieval/vector/dbs

**Reply by dtaivpp** [reply]
Will do! I mostly wanted to make sure that no one else was working on it or maybe there was some reason for it not being implemented.
